### PR TITLE
Exclude federal and international entities from CSA BCC functionality

### LINKED ISSUE #106: Adjustment to CSA BCC Mailing Code for VS Ad-Hoc and Weekly Reports
# 💡 Summary #

Adjust the newly edited code that currently BCC's CSA's on all Vulnerability Scanning ad-hoc and weekly reports to not include specific entity types reports.  

## Motivation and context ##

It has been discussed with the CSA's that 'international' and 'federal' entity reports are not needed to be sent to the CSA's. 

## Implementation notes ##

- This adjusts the code in a way that will no longer produce error messages regarding 'state' data not being entered. 
- This provides the exact reports the CSA's want, and excludes the ones that are, at this time, unnecessary. 
- These are the conditions we are hoping to have added to the current code:
- update the CSA BCC code to exclude agency.type=FEDERAL and agency.type=INTERNATIONAL

## Acceptance criteria ##

- [x] The CSA BCC mailer will only send ad-hoc and weekly reports for entities that are not labeled as 'federal' or 'international' agency types.

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR modifies the CSA BCC functionality implemented in #101 so that CSAs are not sent reports/notifications for federal or international entities.

## 💭 Motivation and context ##

This change was requested by the CyHy team in #106.

Resolves #106.

## 🧪 Testing ##

All automated tests pass.

I did a full dry run (i.e., did everything except actually tell AWS SES to send the email) of sending the CyHy reports and confirmed that the warnings that were previously being generated for federal and international entities are not generated thanks to the changes in this PR.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Add a tag and create a release.
- [x] Create a local Docker image: `docker build -t cisagov/cyhy-mailer:1.7.0 --platform=linux/amd64 .`
- [x] Push Docker image up to Docker Hub: `docker push cisagov/cyhy-mailer:1.7.0`
- [x] Deploy this change to Production (manual process below):
  - [x] On reporter instance `cd /var/cyhy/cyhy-mailer`
  - [x] Update `docker-compose.yml` to use `cisagov/cyhy-mailer:1.7.0`
  - [x] `docker compose pull`
  - [x] Ensure any `--dry-run` flags (used for testing) in composition files are commented out

### CODE REVIEW

**Comment by jsf9k** [comment]
Why not create a release?  It won't hurt anything, will it?

**Comment by dav3r** [comment]
> Why not create a release? It won't hurt anything, will it?

We don't have any releases yet in this repo (only tags), so I wasn't planning to break new ground.

**Comment by jsf9k** [comment]
> > Why not create a release? It won't hurt anything, will it?
> 
> We don't have any releases yet in this repo (only tags), so I wasn't planning to break new ground.

Create a release.  Anytime we create a tag we should create a release to document the changes since the last release.  This repo doesn't have releases already just because it is old.

File: cyhy/mailer/cli.py [hunk 1]
```diff
@@ -793,9 +795,13 @@ def send_cyhy_reports(
...
 
-            # Grab the email address of the appropriate CSA.
+            # Grab the email address of the appropriate CSA for non-Federal and
+            # non-International entities.
             csa_email_address = None
-            if csa_emails is not None:
+            if csa_emails is not None and request["agency"].get("type") not in [
+                "FEDERAL",
+                "INTERNATIONAL",
+            ]:
```
Comment (jsf9k) [inline cyhy/mailer/cli.py]: This list of agency types should be a constant, particularly since we use it twice.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I am requesting one minor improvement.

File: cyhy/mailer/cli.py (hunk 1 above)
Comment (dav3r) [reply cyhy/mailer/cli.py]: Good thinking - please see https://github.com/cisagov/cyhy-mailer/pull/107/commits/2895aafde49529dcdcc5cc0f1ae48360df9e8a36.

**Review by jsf9k — APPROVED** [review]
Strong work!

File: cyhy/mailer/cli.py [hunk 2]
```diff
@@ -793,9 +797,13 @@ def send_cyhy_reports(
             if not to_emails:
                 continue
 
-            # Grab the email address of the appropriate CSA.
+            # Grab the email address of the appropriate CSA for non-Federal and
+            # non-International entities.
```
Comment (mcdonnnj) [inline cyhy/mailer/cli.py]: This matches the case used everywhere else.

```suggestion
            # Grab the email address of the appropriate CSA for non-federal and
            # non-international entities.
```

File: cyhy/mailer/cli.py [hunk 3]
```diff
@@ -1015,9 +1023,13 @@ def send_cyhy_notifications(
         if not to_emails:
             continue
 
-        # Grab the email address of the appropriate CSA.
+        # Grab the email address of the appropriate CSA for non-Federal and
+        # non-International entities.
```
Comment (mcdonnnj) [inline cyhy/mailer/cli.py]: This matches the case used everywhere else.

```suggestion
        # Grab the email address of the appropriate CSA for non-federal and
        # non-international entities.
```

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔ Two suggestions for your consideration.

**Pull request merged**